lib/rust/ensogl/component/grid-view/src/header.rs

```diff
@@ -528,6 +529,46 @@ where
     }
 }
 
+impl<Entry, InnerGridView, HeaderEntry, HeaderModel, HeaderParams> FrpNetworkProvider
+    for GridViewTemplate<Entry, InnerGridView, HeaderEntry, HeaderModel, HeaderParams>
+where
+    InnerGridView: FrpNetworkProvider,
+    HeaderModel: frp::node::Data,
+{
+    fn network(&self) -> &frp::Network {
+        self.model.grid.network()
+    }
+}
+
```

Comment:
I was writing in my recent PR that the bounds are assigned wrongly. This still happens in this PR. Why do you need `HeaderModel: frp::node::Data` here? I'm not reviewing other parts of this PR. Please make sure that ALL introduced bounds are needed.

Reply:
The `HeaderModel: frp::node::Data` bound is required to allow access to `self.model`, due to this same exact bound being present on the definition of the `pub struct GridViewTemplate` (see l.309):

https://github.com/enso-org/enso/blob/42ac5ae3fb5e77bd421c327ceed17f4f45485dca/lib/rust/ensogl/component/grid-view/src/header.rs#L305-L313

As such, the `FrpNetworkProvider` trait implementation we're discussing is rejected by the compiler if the `HeaderModel` bound is removed. Why do you keep insisting it is assigned wrongly? Am I missing something?